impl/src/main/java/org/jboss/weld/bean/proxy/util/WeldDefaultProxyServices.java

```diff
@@ -183,4 +184,25 @@ public final Class<?> publicDefineClass(String name, byte[] b, int off, int len,
             }
         }
     }
+
+    /**
+     * A dummy implementation which has only one purpose - to avoid instantiating {@code DefaultClassFactory.INSTANCE}.
+     * The sole method in this class is never used as we define classes using different means that further vary
+     * between in-container (such as WildFly) and SE setups.
+     *
+     * See {@link WeldDefaultProxyServices#defineClass(Class, String, byte[], int, int)} for details on how we define
+     * classes.
+     */
+    public static class DummyClassFactoryImpl implements ClassFactory {
```

Comment:
I wonder if it would make sense to move this class into `ProxyFactory.java` or even the `org.jboss.weld.bean.proxy` package so that it does not have to be `public`?

Reply:
Sure, I was considering the same. Current location is more or less a remnant of other approaches I tried to fix this issue.
I will make it a separate class under `org.jboss.weld.bean.proxy` and remove the `public` modifier.

Reply:
Done